src/chain/chain-supervision.cc

```diff
@@ -207,6 +209,47 @@ bool PhoneLatticeToProtoSupervision(const SupervisionOptions &opts,
   return true;
 }
 
+bool NormalizePhoneLattice(CompactLattice *clat) {
+  if (clat->Properties(fst::kTopSorted, false) == 0) {
+    if (fst::TopSort(clat) == false) {
+      KALDI_WARN << "Cycles detected in lattice: cannot normalize.";
+      return false;
+    }
+  }
+
+  std::vector<double> beta;
+  if (!ComputeCompactLatticeBetas(*clat, &beta)) {
+    KALDI_WARN << "Failed to compute backward probabilities on lattice.";
+    return false;
+  }
+
+  CompactLattice::Arc::StateId start = clat->Start();  // Should be 0
+  BaseFloat total_backward_cost = beta[start];
```

Comment:
this variable is not a cost, it is a negated cost (or a loglike), so the name isn't suitable.  Fix it where you got the code from, as well.